src/pages/Swap/index.tsx

```diff
@@ -143,13 +143,35 @@ function largerPercentValue(a?: Percent, b?: Percent) {
 
 const TRADE_STRING = 'SwapRouter'
 
-export default function Swap({ className }: { className?: string }) {
-  const navigate = useNavigate()
-  const { account, chainId } = useWeb3React()
+export default function SwapPage({ className }: { className?: string }) {
+  return (
+    <Trace page={InterfacePageName.SWAP_PAGE} shouldLogImpression>
+      <>
+        <PageWrapper>
+          <Swap className={className} />
+          <NetworkAlert />
+        </PageWrapper>
+        <SwitchLocaleLink />
+      </>
+    </Trace>
+  )
+}
+
+export function Swap({
+  className,
+  prefilledState,
+  defaultChainId,
```

Comment:
hm still confused about the mixed usage of `defaultChainId` vs `useWeb3React().chainId` here 

so it looks like some hooks read from `defaultChainId` but some filtering later on is done through `chainId` - can it just read from the passed in prop instead of through the `useWeb3React` hook for consistency?

Reply:
yes, i think we can use the `pageChainId` (renamed from `defaultChainId`) in more places - unfortunately i think we still need references to both, because we need to check whether the user needs to switch chains in their wallet.